controller/deploy_controller.rb

```diff
@@ -4,21 +4,28 @@
 require_relative '../helper/user_comms_helper'
 
 class DeployController
-  def initialize(environ, git_helper, user_comms, develop_tag_generator, other_tag_generator)
-    @environ = environ
+
+  attr_accessor :environ
+  attr_accessor :tag_types
+
+  def initialize(environ, git_helper, user_comms, develop_tag_generator, other_tag_generator, tag_types)
+    self.environ = environ
+    self.tag_types = tag_types
     @git_helper = git_helper
     @user_comms = user_comms
     @develop_tag_generator = develop_tag_generator
     @other_tag_generator = other_tag_generator
   end
 
   def environment_choice
-    @environ.downcase
+    environ.downcase
     @tags_for_this_commit = @git_helper.get_tags_for_this_commit
+    dev_tag = tag_types['first_deploy_step']
```

Comment:
Encapsulate this in a config class, which reads and provides the information of the config file.